### ISSUE TITLE
How to solve systray + minimization problem

### ISSUE THREAD

**Issue by etosan** [herbstluftwm/herbstluftwm#1014]
Due to work and stuffs I started using some software packages that have global hotkeys support and are supposed to go to systray. I run no systray bar, so there is no place for software to go, thus if one sends it close event, instead of "backgrounding" the software terminates and thus stops handling hotkeys.

If one wants the global hotkeys to be handled, software must be running. This is fine, usage wise, because it allows given window to be "brought" into focused monitor/frame, but then this window constantly occupies one frame, still.

What is proper solution for tilers to handle such cases? Would running systray capable bar app solve this?

Would it be possible for herbstlufwm "minimize" (is minimize equal to "unmap"?) window instead, so that it can be brought (with autoraise on urgent or manually somehow) later on without need for systray?

**Comment by t-wissmann** [comment]
> What is proper solution for tilers to handle such cases? Would running systray capable bar app solve this?

Yes, running a bar with systray solves it (e.g. xfce4-panel or lxpanel).

> Would it be possible for herbstlufwm "minimize" (is minimize equal to "unmap"?) window instead, so that it can be brought (with autoraise on urgent or manually somehow) later on without need for systray?

I'd like to implement minimization, but it isn't in hlwm yet.

**Comment by t-wissmann** [comment]
In the current git version (since #1032 / #1033), hlwm supports window minimization! You can minimize through a panel (like lxpanel or xfce4-panel or tint2, ...) or through a keybinding like
```
hc keybind $Mod-Shift-m set_attr clients.focus.minimized true
```
To un-minimize a window, use the bar again or select the window with any other kind of menu (e.g. rofi) or pass its window ID to the `jumpto` command.

**Comment by etosan** [comment]
I will try git master, hopefully, this evenning.

**Comment by federicotorrielli** [comment]
> In the current git version (since #1032 / #1033), hlwm supports window minimization! You can minimize through a panel (like lxpanel or xfce4-panel or tint2, ...) or through a keybinding like
> 
> ```
> hc keybind $Mod-Shift-m set_attr clients.focus.minimized true
> ```
> 
> To un-minimize a window, use the bar again or select the window with any other kind of menu (e.g. rofi) or pass its window ID to the `jumpto` command.

Hi, you should really put this into the documentation! It's very useful!

**Comment by t-wissmann** [comment]
> > In the current git version (since #1032 / #1033), hlwm supports window minimization! You can minimize through a panel (like lxpanel or xfce4-panel or tint2, ...) or through a keybinding like
> > ```
> > hc keybind $Mod-Shift-m set_attr clients.focus.minimized true
> > ```
> > 
> > 
> > To un-minimize a window, use the bar again or select the window with any other kind of menu (e.g. rofi) or pass its window ID to the `jumpto` command.
> 
> Hi, you should really put this into the documentation! It's very useful!

The keybinding is now part of the default autostart.

**Comment by M1ndo** [comment]
@t-wissmann  Would you think about adding a new feature restoring minimized windows through a simple shortcut without using a 3rdparty util to restore it or get the id of the window then pass it to **jumpto** . 
A lot of WM's have this feature and i would like to see it also in herbst.

**Comment by blarz** [comment]
> @t-wissmann Would you think about adding a new feature restoring minimized windows through a simple shortcut without using a 3rdparty util to restore it or get the id of the window then pass it to **jumpto** .
> A lot of WM's have this feature and i would love to see it also in herbst.

Can you check if the newly added keybinding for unminimizing by @apm256 in #1140 does the trick for you?

**Comment by t-wissmann** [comment]
I've also implemented this directly in hlwm in PR #1313, so you can then bind `jumpto last-minimized` and `jumpto longest-minimized`.

**Comment by t-wissmann** [comment]
I think everything from this discussion is now implemented: the minimization and un-minimization is now implemented in hlwm and the [barpyrus panel](https://github.com/t-wissmann/barpyrus) now supports the system tray:

[![image](https://user-images.githubusercontent.com/9048813/143689969-e92fe9ab-9390-4726-bed2-b80192f591e3.png)](https://user-images.githubusercontent.com/9048813/143689969-e92fe9ab-9390-4726-bed2-b80192f591e3.png)

If I missed anything, feel free to re-open this or to open another issue.
